doc/classes/PhysicsShapeQueryParameters2D.xml

```diff
@@ -38,8 +29,11 @@
 		<member name="motion" type="Vector2" setter="set_motion" getter="get_motion" default="Vector2( 0, 0 )">
 			The motion of the shape being queried for.
 		</member>
+		<member name="shape" type="Resource" setter="set_shape" getter="get_shape">
+			The [Shape2D] that will be used for collision/intersection queries. See also [member shape_rid].
+		</member>
 		<member name="shape_rid" type="RID" setter="set_shape_rid" getter="get_shape_rid">
-			The queried shape's [RID]. See also [method set_shape].
+			The queried shape's [RID]. See also [member shape].
```

Comment:
I'm thinking that it would be nice if the documentation explicitly mentioned that `set_shape` will store a reference, and in the `shape_id` property it could say that it's safer to use `shape` if you're creating a `Shape2D` for the query.